## Problem 16: Power Digit Sum

$2^{15} = 32768$ and the sum of its digits is $3 + 2 + 7 + 6 + 8 = 26$.

What is the sum of the digits of the number $2^{1000}$?

In [2]:
# Python can handle large numbers natively.

power = 2**1000
power_string = str(power)
power_string_length = len(power_string)

solution = sum(int(n) for n in power_string)

In [3]:
print(f'There are {power_string_length:,} digits in the number 2^1000 and they sum to {solution:,}.')

There are 302 digits in the number 2^1000 and they sum to 1,366.


### Problem 16 Solution:

There are 302 digits in the number 2^1000 and they sum to 1,366.